pkg/agent/api_docs_test.go

```diff
@@ -0,0 +1,250 @@
+package agent_test
+
+import (
+	"context"
+	"net/url"
+	"testing"
+
+	"github.com/datawire/ambassador/v2/pkg/agent"
+	amb "github.com/datawire/ambassador/v2/pkg/api/getambassador.io/v2"
+	"github.com/datawire/ambassador/v2/pkg/api/getambassador.io/v3alpha1"
+	"github.com/datawire/ambassador/v2/pkg/kates"
+	snapshotTypes "github.com/datawire/ambassador/v2/pkg/snapshot/v1"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestAPIDocsStore(t *testing.T) {
```

Comment:
it would be nice to include a test in which the first `scrape` successfully gets an open api doc from a service, then the next `scrape` does a `GET` that results in a 503 or something from the service, but we make sure we keep the previously fetched docs
(this causes issues in the current devportal and seems fairly easy to regress on)